include/aspect/particle/world.h

```diff
@@ -21,729 +21,310 @@
-          if (found_cell != triangulation->end() && found_cell->point_inside(particle.get_location()))
-            {
-              // If the cell is active, we're at the finest level of refinement and can finish
-              if (found_cell->active())
-                {
-                  particle.set_local(found_cell->is_locally_owned());
-                  return cur_cell;
-                }
-              else
-                {
-                  // Otherwise we need to search deeper
-                  for (child_num=0; child_num<found_cell->n_children(); ++child_num)
-                    {
-                      child_cell = found_cell->child(child_num);
-                      child_li = LevelInd(child_cell->level(), child_cell->index());
-                      res = recursive_find_cell(particle, child_li);
-                      if (res.first != -1 && res.second != -1) return res;
-                    }
-                }
-            }
-
-          // If we still can't find it, return false
-          return LevelInd(-1, -1);
-        };
-
-        /**
-         * Called by listener functions to indicate that the mesh of this
-         * subdomain has changed.
-         */
-        void mesh_changed()
-        {
-          triangulation_changed = true;
-        };
-
       public:
         /**
          * Default World constructor.
          */
-        World()
-        {
-          triangulation_changed = true;
-          total_send = total_recv = 0;
-          world_size = self_rank = 0;
-          num_send = num_recv = send_offset = recv_offset = NULL;
-          send_reqs = recv_reqs = NULL;
-          triangulation = NULL;
-          mapping = NULL;
-          dof_handler = NULL;
-          solution = NULL;
-          integrator = NULL;
-        };
+        World();
 
         /**
-         * Default World destructor, deallocates all relevant arrays and
-         * structures.
+         * Default World destructor.
          */
-        ~World()
+        ~World();
+
+        enum ParticleLoadBalancing
         {
-          if (world_size) MPI_Type_free(&particle_type);
-
-          if (num_send) delete [] num_send;
-          if (num_recv) delete [] num_recv;
-          if (send_offset) delete [] send_offset;
-          if (recv_offset) delete [] recv_offset;
-          if (send_reqs) delete [] send_reqs;
-          if (recv_reqs) delete [] recv_reqs;
+          no_balancing,
+          remove_particles,
+          remove_and_add_particles,
+          repartition
         };
 
         /**
-         * Get the deal.II Mapping associated with this particle world.
+         * Initialize the particle world.
          *
-         * @return The Mapping for this world.
+         * @param [in] new_integrator The new Integrator scheme for this
+         * world.
+         * @param [in] new_manager The new property manager for this
+         * world.
```

Comment:
If you call them `new_*`, are they supposed to *replace* existing objects? Otherwise, maybe drop `new_`?

In general, since these are pointers, who will take over ownership of the objects pointed to?